### DISCUSSION TITLE
1 Beefy Search Node or 3 Virtualized Weaker Search Nodes?

### DISCUSSION THREAD

**Discussion by BuffaloWill** [Security-Onion-Solutions/securityonion#8992] [Unsupported Versions] [answered]
We currently have 1 master node and 3 virtualized search nodes. We are considering moving the 3 virtualized search nodes to a single bare metal system with much better resources. We have a cross cluster setup (i.e. `true_cluster: False`). 

I thought I would ask here as it might help other people in a similar situation:
- Assuming the single bare metal system has better resources, would we see stronger performance by going from three search nodes to one search node? 
- In the Security Onion cross cluster setup, if one search node has a catastrophic failure, is all data on that node lost?

**Comment by isaacgolding** [comment, marked as answer]
So on paper you will see better performance simply because of the 5-15% overhead of VMs vs bare metal.  Having said that I run everything through Proxmox on old beefed up Dell PowerEdge 720 as it made more sense overall for my setup.   In the recent past I had everything split up between master and search nodes and the cross talk between the nodes and searches was my bottle neck.   I would often see 5-10sec delay when requesting info from alerts/dashboards/hunt  So I went to a single very large node on one VM and my overall performance is much better.  

My scenario.   1 location  50-60 computers and another 25+- IOT devices, multiple subnets, Vlans & switches .   A PowerEdge 720 with 8TB of space, 98GB of ram and 2 Processors totaling 32 cores available to the VM.   There are 4 separate Probe interfaces going to 4 separate switches so that I see "everything".

So to answer your first question specifically:   Your situation might be different but I think you will see a performance increase in time saved due to lack of cross talk between nodes and when pulling info, searching etc.

The answer to your second question is yes.   If you loose the search node you loose it data.     Take a look at: https://docs.securityonion.net/en/2.3/elasticsearch.html?highlight=elastic#cross-cluster-search

The docs also mention that if you are using a "forward node" nothing is stored locally and filebeat sends it all to the manager node.

**Comment by BuffaloWill** [comment]
Thank you @isaacgolding! This is really helpful information. I am starting to lean towards a single search node due to the exact reasons you mentioned.
